<div style="width: 100%; clear: both;">
<div style="float: left; width: 50%;">
<img src="http://www.uoc.edu/portal/_resources/common/imatges/marca_UOC/UOC_Masterbrand.jpg" align="left">
</div>
<div style="float: right; width: 50%;">
<p style="margin: 0; padding-top: 22px; text-align:right;">GABRIEL PALADINES HERRERA</p>
<p style="margin: 0; padding-top: 22px; text-align:right;">M2.859 · Visualización de datos · PEC2</p>
<p style="margin: 0; text-align:right;">2019-2 · Máster universitario en Ciencia de datos (Data science)</p>
<p style="margin: 0; text-align:right; padding-button: 100px;">Estudios de Informática, Multimedia y Telecomunicación

 </p>
</div>
</div>
<div style="width:100%;">&nbsp;</div>


# PEC 2, PEC 3 Y PEC 4: Familiarízate con los datos


Sobre el proyecto:


En esta PEC2 conoceréis los datos que os proponemos y haréis visualizaciones de los datos, buscando tendencias y evidencias. En la PEC3 redactaréis  el informe de proyecto donde se describen las partes del proyecto y se explica cómo se hará cada elemento; tanto en cuanto a los datos, como en cuanto al diseño. A la última PEC4 se crea la visualización final y lo prepara todo para la entrega de proyecto.
   
Para eso necesitaremos las siguientes librerías:

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
from sklearn import datasets
from sklearn.manifold import TSNE
from sklearn.decomposition import PCA
from sklearn.preprocessing import StandardScaler

import matplotlib.pyplot as plt


pd.set_option('display.max_columns', None)

%matplotlib inline

# 1. Carga del conjunto de datos

En primer lugar, debéis cargar el conjunto de datos (más información en el enlace [https://www.peaceagreements.org/search](https://www.peaceagreements.org/search)).

In [2]:
data = pd.read_csv('pax_all_agreements_data_v2.csv')
data.head()

Con       Contp   PP                                        PPName  \
0  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
1  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
2  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
3  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
4  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   

                  Reg  AgtId  \
0  Europe and Eurasia   1739   
1  Europe and Eurasia   1923   
2  Europe and Eurasia    864   
3  Europe and Eurasia    848   
4  Europe and Eurasia    849   

                                                 Agt         Dat  \
0  Agreement between the Islamic Republic of Afgh...  2016-09-22   
1  Agreement between the two campaign teams regar...  2014-09-21   
2  Tokyo Declaration Partnership for Self-Relianc...  2012-07-08   
3  Conclusions of the Conference on Afghanistan a...  2011-12-05   
4  Istanbul Process on Regional Security and Coop...  2011-11-02   

                     Status  Lgt  N_characters        Agtp   Stage StageSub  \
0  Multiparty signed/agreed    4         10203       Intra  SubPar  MultIss   
1  Multiparty signed/agreed    4         11110       Intra     Imp   ExtSub   
2  Multiparty signed/agreed   14         39746  InterIntra     Imp   ExtSub   
3  Multiparty signed/agreed    7         24106  InterIntra     Pre     Prin   
4  Multiparty signed/agreed   10         21299  InterIntra     Imp   ExtSub   

                                                Part  \
0  President of the IRA Mohammad Ashraf Ghani, ch...   
1  Dr. Mohammad Ashraf Ghani Ahmadzai \r\nDr. Abd...   
2  Not signed, agreement mentions the following p...   
3  Not signed, agreement mentions the following p...   
4  Not signed, agreements mentions the following ...   

                                            ThrdPart  \
0                                                NaN   
1  The foregoing signatures were witnessed by:\r\...   
2  Ambassador Kazuo Kodama, Chargé d’affaires ad ...   
3                                                NaN   
4  Not signed, agreement mentions the following p...   

                                              OthAgr Loc1ISO Loc2ISO  \
0                                                NaN     AFG     NaN   
1  Page 3, D. Creation of the position of leader ...     AFG     NaN   
2  Page 2, Preamble, 1\r\n...\r\nThese undertakin...     AFG     NaN   
3  Page 1, 5\r\n...\r\nReaffirming our commitment...     AFG     NaN   
4  Page 1, 2\r\nWe affirm the commitments enshrin...     AFG     NaN   

   Loc1GWNO  Loc2GWNO UcdpCon  UcdpAgr  PamAgr  CowWar  GCh  GChRhet  \
0     700.0       NaN     333   1488.0     NaN   851.0    0        0   
1     700.0       NaN     333      NaN     NaN   225.0    2        1   
2     700.0       NaN     333      NaN     NaN   225.0    1        1   
3     700.0       NaN     333      NaN     NaN   225.0    1        1   
4     700.0       NaN     333      NaN     NaN   225.0    0        0   

   GChAntid  GChSubs  GChOth  GDis  GDisRhet  GDisAntid  GDisSubs  GDisOth  \
0         0        0       0     0         0          0         0        0   
1         0        1       0     2         0          0         1        0   
2         0        0       0     0         0          0         0        0   
3         0        0       0     0         0          0         0        0   
4         0        0       0     0         0          0         0        0   

   GAge  GAgeRhet  GAgeAntid  GAgeSubs  GAgeOth  GMig  GMigRhet  GMigAntid  \
0     0         0          0         0        0     0         0          0   
1     0         0          0         0        0     0         0          0   
2     0         0          0         0        0     0         0          0   
3     0         0          0         0        0     0         0          0   
4     0         0          0         0        0 

# Analisis Exploratorio

In [25]:
data.isnull().any()

Con             False
Contp           False
PP               True
PPName           True
Reg             False
AgtId           False
Agt             False
Dat             False
Status          False
Lgt             False
N_characters    False
Agtp            False
Stage           False
StageSub         True
Part             True
ThrdPart         True
OthAgr           True
Loc1ISO          True
Loc2ISO          True
Loc1GWNO         True
Loc2GWNO         True
UcdpCon          True
UcdpAgr          True
PamAgr           True
CowWar           True
GCh             False
GChRhet         False
GChAntid        False
GChSubs         False
GChOth          False
                ...  
SsrPsf          False
SsrFf           False
Cor             False
SsrCrOcr        False
SsrDrugs        False
Terr            False
TjGen           False
TjAm            False
TjAmPro         False
TjSan           False
TjPower         False
TjAmBan         False
TjCou           False
TjJaNc          False
TjJaIc    

In [26]:
data.describe()

PP        AgtId          Lgt   N_characters     Loc1GWNO  \
count  1782.000000  1789.000000  1789.000000    1789.000000  1769.000000   
mean     89.603255  1106.604807     6.359978   11959.409726   528.292256   
std      47.323516   635.813541    17.493402   37567.806638   227.494776   
min       1.000000     1.000000     1.000000       0.000000    41.000000   
25%      51.000000   547.000000     1.000000    2389.000000   359.000000   
50%      94.000000  1089.000000     2.000000    4362.000000   540.000000   
75%     120.000000  1671.000000     5.000000    9443.000000   678.000000   
max     203.000000  2231.000000   323.000000  908459.000000   940.000000   

         Loc2GWNO       UcdpAgr     PamAgr       CowWar          GCh  \
count  232.000000  2.970000e+02  33.000000  1111.000000  1789.000000   
mean   439.159483  2.903701e+10  17.272727   760.739874     0.313583   
std    197.221974  3.602522e+11  10.022702   253.699787     0.773252   
min      2.000000  2.400000e+01   1.000000   139.000000     0.000000   
25%    345.000000  1.107000e+03   9.000000   817.000000     0.000000   
50%    371.000000  1.334000e+03  17.000000   857.000000     0.000000   
75%    625.000000  1.523000e+03  26.000000   905.000000     0.000000   
max    770.000000  5.171991e+12  34.000000   941.000000     3.000000   

           GChRhet     GChAntid      GChSubs       GChOth         GDis  \
count  1789.000000  1789.000000  1789.000000  1789.000000  1789.000000   
mean      0.078815     0.003354     0.104528     0.008385     0.140861   
std       0.269525     0.057831     0.306029     0.091208     0.560891   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     3.000000   

          GDisRhet    GDisAntid     GDisSubs      GDisOth         GAge  \
count  1789.000000  1789.000000  1789.000000  1789.000000  1789.000000   
mean      0.021241     0.010061     0.049189     0.002795     0.093348   
std       0.144227     0.099829     0.216324     0.052807     0.452071   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     3.000000   

          GAgeRhet    GAgeAntid     GAgeSubs  GAgeOth         GMig  \
count  1789.000000  1789.000000  1789.000000   1789.0  1789.000000   
mean      0.018446     0.007826     0.030184      0.0     0.027390   
std       0.134595     0.088140     0.171142      0.0     0.247649   
min       0.000000     0.000000     0.000000      0.0     0.000000   
25%       0.000000     0.000000     0.000000      0.0     0.000000   
50%       0.000000     0.000000     0.000000      0.0     0.000000   
75%       0.000000     0.000000     0.000000      0.0     0.000000   
max       1.000000     1.000000     1.000000      0.0     3.000000   

          GMigRhet    GMigAntid     GMigSubs  GMigOth          GRa  \
count  1789.000000  1789.000000  1789.000000   1789.0  1789.000000   
mean      0.002795     0.000559     0.010061      0.0     0.377306   
std       0.052807     0.023643     0.099829      0.0     0.839569   
min       0.000000     0.000000     0.000000      0.0     0.000000   
25%       0.000000     0.000000     0.000000      0.0     0.000000   
50%       0.000000     0.000000     0.000000      0.0     0.000000   
75%       0.000000     0.000000     0.000000      0.0     0.000000   
max       1.000000     1.000000     1.000000      0.0     3.000000   

           GRaRhet     GRaAntid      GRaSubs       GRaOth          GRe  \
count  1789.

In [27]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1789 entries, 0 to 1788
Columns: 266 entries, Con to ImSrc
dtypes: float64(6), int64(244), object(16)
memory usage: 3.6+ MB


In [28]:
cat_data = data.select_dtypes(include=['object']).copy()
cat_data.head()

Con       Contp                                        PPName  \
0  Afghanistan  Government  Afghanistan: 2000s Post-intervention process   
1  Afghanistan  Government  Afghanistan: 2000s Post-intervention process   
2  Afghanistan  Government  Afghanistan: 2000s Post-intervention process   
3  Afghanistan  Government  Afghanistan: 2000s Post-intervention process   
4  Afghanistan  Government  Afghanistan: 2000s Post-intervention process   

                  Reg                                                Agt  \
0  Europe and Eurasia  Agreement between the Islamic Republic of Afgh...   
1  Europe and Eurasia  Agreement between the two campaign teams regar...   
2  Europe and Eurasia  Tokyo Declaration Partnership for Self-Relianc...   
3  Europe and Eurasia  Conclusions of the Conference on Afghanistan a...   
4  Europe and Eurasia  Istanbul Process on Regional Security and Coop...   

          Dat                    Status        Agtp   Stage StageSub  \
0  2016-09-22  Multiparty signed/agreed       Intra  SubPar  MultIss   
1  2014-09-21  Multiparty signed/agreed       Intra     Imp   ExtSub   
2  2012-07-08  Multiparty signed/agreed  InterIntra     Imp   ExtSub   
3  2011-12-05  Multiparty signed/agreed  InterIntra     Pre     Prin   
4  2011-11-02  Multiparty signed/agreed  InterIntra     Imp   ExtSub   

                                                Part  \
0  President of the IRA Mohammad Ashraf Ghani, ch...   
1  Dr. Mohammad Ashraf Ghani Ahmadzai \r\nDr. Abd...   
2  Not signed, agreement mentions the following p...   
3  Not signed, agreement mentions the following p...   
4  Not signed, agreements mentions the following ...   

                                            ThrdPart  \
0                                                NaN   
1  The foregoing signatures were witnessed by:\r\...   
2  Ambassador Kazuo Kodama, Chargé d’affaires ad ...   
3                                                NaN   
4  Not signed, agreement mentions the following p...   

                                              OthAgr Loc1ISO Loc2ISO UcdpCon  
0                                                NaN     AFG     NaN     333  
1  Page 3, D. Creation of the position of leader ...     AFG     NaN     333  
2  Page 2, Preamble, 1\r\n...\r\nThese undertakin...     AFG     NaN     333  
3  Page 1, 5\r\n...\r\nReaffirming our commitment...     AFG     NaN     333  
4  Page 1, 2\r\nWe affirm the commitments enshrin...     AFG     NaN     333

In [29]:
print(cat_data.isnull().values.sum())

3519


In [30]:
print(cat_data.isnull().sum())

Con            0
Contp          0
PPName         7
Reg            0
Agt            0
Dat            0
Status         0
Agtp           0
Stage          0
StageSub      29
Part          15
ThrdPart     869
OthAgr       852
Loc1ISO       44
Loc2ISO     1557
UcdpCon      146
dtype: int64


In [31]:
cat_data = cat_data.fillna(cat_data['Loc1ISO'].value_counts().index[0])

In [32]:
print(cat_data['Con'].value_counts())

Bosnia and Herzegovina/Yugoslavia (former)                                                                                                                             133
Colombia                                                                                                                                                               128
Philippines/Mindanao                                                                                                                                                   126
Syria                                                                                                                                                                   82
Nepal                                                                                                                                                                   51
Yemen                                                                                                                                            

In [33]:
print(cat_data['Contp'].value_counts())

Government/territory    813
Government              662
Inter-group             206
Territory               107
Other                     1
Name: Contp, dtype: int64


In [34]:
print(cat_data['Reg'].value_counts())

Africa (excl MENA)              529
Europe and Eurasia              407
Asia and Pacific                376
Middle East and North Africa    248
Americas                        195
Cross-regional                   34
Name: Reg, dtype: int64


In [35]:
print(cat_data['Status'].value_counts())

Multiparty signed/agreed            1652
Unilateral document                  120
Status unclear                        10
Agreement with subsequent status       7
Name: Status, dtype: int64


In [36]:
print(cat_data['Agtp'].value_counts())

Intra         1293
InterIntra     234
IntraLocal     190
Inter           72
Name: Agtp, dtype: int64


In [38]:
cat_data['Status'] = cat_data['Status'].astype('category')
cat_data['Agtp'] = cat_data['Agtp'].astype('category')
cat_data['Contp'] = cat_data['Contp'].astype('category')
cat_data['Stage'] = cat_data['Stage'].astype('category')

print(cat_data.dtypes)

Con           object
Contp       category
PPName        object
Reg           object
Agt           object
Dat           object
Status      category
Agtp        category
Stage       category
StageSub      object
Part          object
ThrdPart      object
OthAgr        object
Loc1ISO       object
Loc2ISO       object
UcdpCon       object
dtype: object


# Reduccion de Dimensiones

A continuacion seleccionaremos todas los atributos de informacion basica y los atritbutos relacionados a los derechos humanos de acuerdo al diccionario de datos (https://www.peaceagreements.org/files/PA_X_codebook_Version2_June_2019.pdf)

Listado de atributos que contendra nuestro dataset final:

**Basic information**
- Con: Country/Entity
- Contp: Conflict type
- PP_name: Peace process Name 
- PP: Peace Process Number
- Reg: Region
- AgtId: Agreement ID
- Agt: Agreement Name
- Dat: Date Signed
- Status: Agreement Definition and Status
- Lgt: Agreement length (pages)
- N_characters: Agreement length (characters)
- Agtp: Agreement/conflict type
- Stage: Agreement stage 
- StageSub: subcoding of stage Parties
- ThrdPart: Third Parties
- OthAgr: Other Agreement
- Loc1ISO
- Loc2ISO
- Loc1GWNO
- Loc2GWNO
- UcdpCon
- UcdpAgr
- PamAgr
- CowWar

**GROUPS**

GENDER
- GeWom: Women, girls and gender
- GeMe: Men and Boys
- GeMeNu: Gender neutrality wording
- GeMeOth: other
- GeSo: Sexual Orientation 
- GeLgbti: LGBTI references
- GeFa: Family

HUMAN RIGHTS AND EQUALITY
- **HrGen: Human Rights/ Rule of Law**
- **EqGen: Equality** 
- **HrDem: Democracy**
- **Prot: Protection measures**
- Prot Civ: Protection of civilians
- ProtGrp: Protection of groups
- ProtLgl: Protection of rights and legal frameworks
- ProtOth: Other protection measures
- **HrFra: Human Rights Framework**
- HrfSp: Isolated rights
- HrfBor: Bill of Rights
- HrfTinc: Incorporation of human rights treaties, humanitarian law, or international criminal law
- treaties (Treaty Incorporation)
- HrfOth: Other
- **HrCp: Civil and political rights** 
- CprLife: Life
- CprTort: Torture 
- CprEq: Equality 
- CprSlav: Slavery
- CprLib: Liberty and security of person
- CprDet: Humane treatment in detention
- CprFmov: Freedom of movement
- CprFspe: Freedom of speech
- CprFass: Freedom of association
- CprTria: Fair trial
- CprPriv: Privacy and family life
- CprVote: Vote and take part 
- CprReli: Thought, opinion, conscience and religion 
- CprOth: Other
- **HrSec: Socio-economic rights**
- SerProp: Property
- SerWork: Work
- SerHeal: Health
- SerEdu: Education 
- SerStdl: Adequate standard of living
- SerShel: Shelter/housing 
- SerSs: Social security 
- SerCult: Cultural rights 
- SerOth: Other
- **HrNi: NHRI (National Human Rights Institution)** 
- HrNiMe: Mentions of NHRI 
- HrNiNe: New or fundamentally revised NHRI 
- HrNiOth: Other NHRI
- **HrIi: Regional or international human rights institutions**
- HrIiMon: Monitoring calls
- HrIiBod: Body tasked
- HrIiOth: Other regional or international human rights institutions
- **HrMob: mobility/access**
- **HrDet: Detention Procedures**
- **Med: Media and communication**
- MedGov: Governance of Media. 
- MedSubs: Media Roles
- MedLog: Media Logistics
- MedOth: Media Other
- **HrCit: Citizenship**
- CitGen: Citizen, general
- CitRights: Citizens, specific rights
- CitDef: Citizenship delimitation
- CitOth: Citizenship other

In [16]:
basic_columns = ['Con', 'Contp', 'PP_name', 'PP', 'Reg', 'AgtId', 'Agt', 'Dat','Status','Lgt', 'N_characters', 'Agtp', 'Stage', 'StageSub', 'ThrdPart', 'OthAgr', 'Loc1ISO', 'Loc2ISO', 'UcdpCon', 'UcdpAgr', 'PamAgr', 'CowWar']
gender_columns = ['GeWom', 'GeMe', 'GeMeOth', 'GeSo', 'GeLgbti', 'GeFa']
hr_columns = ['HrGen', 'EqGen', 'HrDem', 'Prot', 'HrFra', 'HrCp', 'HrNi', 'Hrli', 'HrMob', 'HrMob', 'HrDet', 'Med', 'HrCit']
selected_data = data.loc[:, basic_columns + gender_columns + hr_columns]

In [17]:
selected_data.head()

Con       Contp  PP_name   PP                 Reg  AgtId  \
0  Afghanistan  Government      NaN  2.0  Europe and Eurasia   1739   
1  Afghanistan  Government      NaN  2.0  Europe and Eurasia   1923   
2  Afghanistan  Government      NaN  2.0  Europe and Eurasia    864   
3  Afghanistan  Government      NaN  2.0  Europe and Eurasia    848   
4  Afghanistan  Government      NaN  2.0  Europe and Eurasia    849   

                                                 Agt         Dat  \
0  Agreement between the Islamic Republic of Afgh...  2016-09-22   
1  Agreement between the two campaign teams regar...  2014-09-21   
2  Tokyo Declaration Partnership for Self-Relianc...  2012-07-08   
3  Conclusions of the Conference on Afghanistan a...  2011-12-05   
4  Istanbul Process on Regional Security and Coop...  2011-11-02   

                     Status  Lgt  N_characters        Agtp   Stage StageSub  \
0  Multiparty signed/agreed    4         10203       Intra  SubPar  MultIss   
1  Multiparty signed/agreed    4         11110       Intra     Imp   ExtSub   
2  Multiparty signed/agreed   14         39746  InterIntra     Imp   ExtSub   
3  Multiparty signed/agreed    7         24106  InterIntra     Pre     Prin   
4  Multiparty signed/agreed   10         21299  InterIntra     Imp   ExtSub   

                                            ThrdPart  \
0                                                NaN   
1  The foregoing signatures were witnessed by:\r\...   
2  Ambassador Kazuo Kodama, Chargé d’affaires ad ...   
3                                                NaN   
4  Not signed, agreement mentions the following p...   

                                              OthAgr Loc1ISO Loc2ISO UcdpCon  \
0                                                NaN     AFG     NaN     333   
1  Page 3, D. Creation of the position of leader ...     AFG     NaN     333   
2  Page 2, Preamble, 1\r\n...\r\nThese undertakin...     AFG     NaN     333   
3  Page 1, 5\r\n...\r\nReaffirming our commitment...     AFG     NaN     333   
4  Page 1, 2\r\nWe affirm the commitments enshrin...     AFG     NaN     333   

   UcdpAgr  PamAgr  CowWar  GeWom  GeMe  GeMeOth  GeSo  GeLgbti  GeFa  HrGen  \
0   1488.0     NaN   851.0      1     0        0     0        0     0      0   
1      NaN     NaN   225.0      1     0        0     0        0     0      1   
2      NaN     NaN   225.0      1     1        0     0        0     0      1   
3      NaN     NaN   225.0      1     1        0     0        0     0      1   
4      NaN     NaN   225.0      1     0        0     0        0     0      1   

   EqGen  HrDem  Prot  HrFra  HrCp  HrNi  Hrli  HrMob  HrMob  HrDet  Med  \
0      1      0     0      0     0     0   NaN      0      0      0    0   
1      0      0     0      0     0     0   NaN      0      0      0    0   
2      1      2     1      3     0     1   NaN      0      0      0    1   
3      1      2     1      0     0     0   NaN      0      0      0    0   
4      0      1     0      0     0     0   NaN      1      1      0    1   

   HrCit  
0      0  
1      0  
2      0  
3      2  
4      0

**Nota:**
Para los campos que son de las categorías genero y derechos humanos, debemos recalcar que el valor 0 indica la no presencia de este atributo en el acuerdo de paz y mayor a 0 la presencia de este atributo en el acuerdo de paz. Para algunos de estos atributos presentan valores que indican 1, 2 o 3 esto significa la inclusión de una subcategoría especifica a ese atributo o el grado de incidencia de este atributo que tuvo en el acuerdo de paz. Para nuestro estudio nos centraremos en el primer nivel que solo indica la presencia o no de este atributo.

In [18]:
selected_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1789 entries, 0 to 1788
Data columns (total 41 columns):
Con             1789 non-null object
Contp           1789 non-null object
PP_name         0 non-null float64
PP              1782 non-null float64
Reg             1789 non-null object
AgtId           1789 non-null int64
Agt             1789 non-null object
Dat             1789 non-null object
Status          1789 non-null object
Lgt             1789 non-null int64
N_characters    1789 non-null int64
Agtp            1789 non-null object
Stage           1789 non-null object
StageSub        1760 non-null object
ThrdPart        920 non-null object
OthAgr          937 non-null object
Loc1ISO         1745 non-null object
Loc2ISO         232 non-null object
UcdpCon         1643 non-null object
UcdpAgr         297 non-null float64
PamAgr          33 non-null float64
CowWar          1111 non-null float64
GeWom           1789 non-null int64
GeMe            1789 non-null int64
GeMeOth       

**De acuerdo al informe anterior vemos que las columnas PP_name y Hrli son nulas así que procederemos a eliminarlas de nuestro dataset, así como el atributo PamAgr solo contiene 33 registros con datos.**

In [19]:
selected_data = selected_data.drop(['PP_name', 'Hrli', 'PamAgr'], axis=1)
selected_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1789 entries, 0 to 1788
Data columns (total 38 columns):
Con             1789 non-null object
Contp           1789 non-null object
PP              1782 non-null float64
Reg             1789 non-null object
AgtId           1789 non-null int64
Agt             1789 non-null object
Dat             1789 non-null object
Status          1789 non-null object
Lgt             1789 non-null int64
N_characters    1789 non-null int64
Agtp            1789 non-null object
Stage           1789 non-null object
StageSub        1760 non-null object
ThrdPart        920 non-null object
OthAgr          937 non-null object
Loc1ISO         1745 non-null object
Loc2ISO         232 non-null object
UcdpCon         1643 non-null object
UcdpAgr         297 non-null float64
CowWar          1111 non-null float64
GeWom           1789 non-null int64
GeMe            1789 non-null int64
GeMeOth         1789 non-null int64
GeSo            1789 non-null int64
GeLgbti      

**Como resultado de la reducción de dimesiones hemos pasado de 266 atributos a 38.**

# Tranformaciones

Previo a la generación del conjunto de datos final debemos realizar ciertas transformaciones:

- Atributo Con: Algunos valores presentan mas de un país separados por el carácter ‘/’, en este caso como nuestra intención es representar un mapa de países, transformaremos el valor de este atributo seleccionado el primer país presente y los restantes los colocaremos en un nuevo atributo llamando Con2.

In [27]:
selected_data["Con"]= selected_data["Con"].str.split("/", n = 1, expand = True) 
selected_data.head()

Con       Contp   PP                 Reg  AgtId  \
0  Afghanistan  Government  2.0  Europe and Eurasia   1739   
1  Afghanistan  Government  2.0  Europe and Eurasia   1923   
2  Afghanistan  Government  2.0  Europe and Eurasia    864   
3  Afghanistan  Government  2.0  Europe and Eurasia    848   
4  Afghanistan  Government  2.0  Europe and Eurasia    849   

                                                 Agt         Dat  \
0  Agreement between the Islamic Republic of Afgh...  2016-09-22   
1  Agreement between the two campaign teams regar...  2014-09-21   
2  Tokyo Declaration Partnership for Self-Relianc...  2012-07-08   
3  Conclusions of the Conference on Afghanistan a...  2011-12-05   
4  Istanbul Process on Regional Security and Coop...  2011-11-02   

                     Status  Lgt  N_characters        Agtp   Stage StageSub  \
0  Multiparty signed/agreed    4         10203       Intra  SubPar  MultIss   
1  Multiparty signed/agreed    4         11110       Intra     Imp   ExtSub   
2  Multiparty signed/agreed   14         39746  InterIntra     Imp   ExtSub   
3  Multiparty signed/agreed    7         24106  InterIntra     Pre     Prin   
4  Multiparty signed/agreed   10         21299  InterIntra     Imp   ExtSub   

                                            ThrdPart  \
0                                                NaN   
1  The foregoing signatures were witnessed by:\r\...   
2  Ambassador Kazuo Kodama, Chargé d’affaires ad ...   
3                                                NaN   
4  Not signed, agreement mentions the following p...   

                                              OthAgr Loc1ISO Loc2ISO UcdpCon  \
0                                                NaN     AFG     NaN     333   
1  Page 3, D. Creation of the position of leader ...     AFG     NaN     333   
2  Page 2, Preamble, 1\r\n...\r\nThese undertakin...     AFG     NaN     333   
3  Page 1, 5\r\n...\r\nReaffirming our commitment...     AFG     NaN     333   
4  Page 1, 2\r\nWe affirm the commitments enshrin...     AFG     NaN     333   

   UcdpAgr  CowWar  GeWom  GeMe  GeMeOth  GeSo  GeLgbti  GeFa  HrGen  EqGen  \
0   1488.0   851.0      1     0        0     0        0     0      0      1   
1      NaN   225.0      1     0        0     0        0     0      1      0   
2      NaN   225.0      1     1        0     0        0     0      1      1   
3      NaN   225.0      1     1        0     0        0     0      1      1   
4      NaN   225.0      1     0        0     0        0     0      1      0   

   HrDem  Prot  HrFra  HrCp  HrNi  HrMob  HrMob  HrDet  Med  HrCit  
0      0     0      0     0     0      0      0      0    0      0  
1      0     0      0     0     0      0      0      0    0      0  
2      2     1      3     0     1      0      0      0    1      0  
3      2     1      0     0     0      0      0      0    0      2  
4      1     0      0     0     0      1      1      0    1      0

# Generación de dataset final

**A continuación generaremos el archivo final que será importado en Tableau.**

In [29]:
selected_data.to_csv(r'pec4_dataset.csv', sep=';')